### PR TITLE
Add 3 new syscalls: spawn, get_memory_limit and set_content

### PR DESCRIPTION
**Pull request by mohanson**
Note: 
1) these syscalls only avaliable after ckb2023, 
2) and simulator is not currently implemented.

### CODE REVIEW

File: src/syscalls/native.rs [hunk 1]
```diff
@@ -581,3 +581,66 @@ pub fn exec(
...
+    pub content: *mut u8,
+    pub content_length: *mut u64,
+}
+
+/// The Spawn and the latter two syscalls: Get Memory Limit and Set Content
+/// together, implement a way to call another CKB Script in a CKB Script.
+/// Note: available after ckb2023.
+///
+/// Returns success or a syscall error.
+pub fn spawn(index: usize, source: Source, bounds: usize, argv: &[&CStr], spgs: &SpawnArgs) -> u64 {
```
Comment (XuJiandong) [inline src/syscalls/native.rs:602]: In high_level api, we need a more user friendly one, like this:
```
pub fn spawn(code_hash: &[u8], hash_type: ScriptHashType, argv: & Vec<String>, exit_code: &mut u8, content: &mut Vec<u8>, memory_limit: usize);
```
Adjust it if needed. Note, the size of `content` will be shrinked after call.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Comment by jjyr** [comment]
Please use `feature` to control those syscalls.

``` rust
#[cfg(features = "ckb2023")]
pub use ckb2023::*;

mod ckb2023 {
  pub fn spawn() { .. }
}
```

File: src/syscalls/native.rs [hunk 2]
```diff
@@ -581,3 +581,66 @@ pub fn exec(
...
+///
+/// Returns a number between 1 and 8, representing 0.5 to 4M of memory.
+pub fn get_memory_limit() -> u64 {
+    unsafe { syscall(0, 0, 0, 0, 0, 0, 0, SYS_GET_MEMORY_LIMIT) }
+}
+
+/// Set content.
+/// Note: available after ckb2023.
+///
+/// Return the actual written data length or a syscall error.
```
Comment (XuJiandong) [inline src/syscalls/native.rs:634]: Add more comments: when content length given at `spawn` is smaller than size of `buf` here, it returns smaller one.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Comment by blckngm** [comment]
If we use (non-default) features, we should use

```
[package.metadata.docs.rs]
all-features = true
```

so that docs.rs will have docs for them.

**Review by XuJiandong — APPROVED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -8,15 +8,19 @@ repository = "https://github.com/nervosnetwork/ckb-std"
...
 exclude = ["docs"]
 
+[package.metadata.docs.rs]
+all-features = true
+
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [features]
-default = ["allocator", "ckb-types", "dlopen-c", "dummy-libc"]
+default = ["allocator", "ckb-types", "dlopen-c", "dummy-libc", "ckb2023"]
```
Comment (jjyr) [inline Cargo.toml]: We should require users to explicitly turn on the 'ckb2023' feature until the mainnet hardfork.

File: src/ckb_constants.rs [hunk 4]
```diff
@@ -16,6 +16,9 @@ pub const SYS_LOAD_INPUT_BY_FIELD: u64 = 2083;
 pub const SYS_LOAD_CELL_DATA_AS_CODE: u64 = 2091;
 pub const SYS_LOAD_CELL_DATA: u64 = 2092;
 pub const SYS_DEBUG: u64 = 2177;
+pub const SYS_SPAWN: u64 = 2101;
```
Comment (jjyr) [inline src/ckb_constants.rs:20]: Please make those syscall numbers only available with ckb2023 feature

**Review by jjyr — CHANGES_REQUESTED** [review]

**Comment by blckngm** [comment]
And there's the `doc_auto_cfg` feature for automatically generating `only available on crate feature ...` annotations. We should enable this on docs.rs.

**Comment by mohanson** [comment]
> And there's the `doc_auto_cfg` feature for automatically generating `only available on crate feature ...` annotations. We should enable this on docs.rs.

doc_auto_cfg  may not be used on the stable release channel

**Review by jjyr — APPROVED** [review]

**Comment by blckngm** [comment]
> doc_auto_cfg may not be used on the stable release channel

Docs.rs uses nightly. I'll submit a PR for this.

**Pull request merged**